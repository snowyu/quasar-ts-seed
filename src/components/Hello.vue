<!-- src/components/Hello.vue -->
<template>
  <q-page class="flex flex-center">
    <q-list no-border>
      <q-item class="greeting">Hello {{title}}{{exclamationMarks}}</q-item>
      <q-btn @click="decrement">-</q-btn>
      <q-btn @click="increment">+</q-btn>
    </q-list>
  </q-page>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
    name: 'Hello',
    props: {title: {type: String}, initialEnthusiasm: {default: 1, type: Number}},
    data() {
        return {
            enthusiasm: this.initialEnthusiasm,
        }
    },
    methods: {
        increment() { this.enthusiasm++; },
        decrement() {
            if (this.enthusiasm > 1) {
                this.enthusiasm--;
            }
        },
    },
    computed: {
        exclamationMarks(): string {
            return Array(this.enthusiasm + 1).join('!');
        }
    }
});
</script>
