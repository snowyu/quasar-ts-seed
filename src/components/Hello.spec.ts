import { shallowMount } from '@vue/test-utils';
import Hello from './Hello.vue';

describe('Hello.vue', () => {
  it('renders name test', () => {
    const l = 1;
    const wrapper = shallowMount(Hello, {
      propsData: { title: 'test' }
    });
    const vGreeting = wrapper.find('.greeting');
    expect(vGreeting.text()).toEqual('Hello test!');
  })

  it('matches snapshot', () => {
    const name = 'snap';
    let wrapper = shallowMount(Hello, {
      propsData: { title: name }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
