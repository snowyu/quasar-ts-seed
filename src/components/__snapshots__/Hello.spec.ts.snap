// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello.vue matches snapshot 1`] = `
<q-list no-border="">
  <q-item class="greeting">Hello snap!</q-item>
  <q-btn id="btnDec">-</q-btn>
  <q-btn id="btnInc">+</q-btn>
</q-list>
`;
